addons/point_of_sale/models/pos_session.py

```diff
@@ -237,19 +237,29 @@ def action_pos_session_open(self):
 
     @api.multi
     def action_pos_session_closing_control(self):
+        self._check_pos_session_balance()
         for session in self:
             #DO NOT FORWARD-PORT
             if session.state == 'closing_control':
                 session.action_pos_session_close()
                 continue
-            for statement in session.statement_ids:
-                if (statement != session.cash_register_id) and (statement.balance_end != statement.balance_end_real):
-                    statement.write({'balance_end_real': statement.balance_end})
             session.write({'state': 'closing_control', 'stop_at': fields.Datetime.now()})
             if not session.config_id.cash_control:
                 session.action_pos_session_close()
 
     @api.multi
+    def _check_pos_session_balance(self):
+        for session in self:
+            for statement in session.statement_ids:
+                if (statement != session.cash_register_id) and (statement.balance_end != statement.balance_end_real):
+                    statement.write({'balance_end_real': statement.balance_end})
+
+    @api.multi
+    def action_pos_session_validate(self):
```

Comment:
the method `action_pos_session_validate` does not exists in v10. In v10 it was `action_pos_session_closing_control` so maybe, we should only move the computation _before_ the `if session.state == 'closing_control':` (to test)